### PR TITLE
[google_maps_flutter] Adds support for circle overlays to the Google Maps plugin

### PR DESCRIPTION
**Pull request by matthewlloyd**
## Description

Adds support for circle overlays to the Google Maps plugin

This is relatively trivial, requiring only some additional logic
to disambiguate click events between the various possible overlays.

I've followed the majority (2/3: Android and JavaScript) Google Maps
API convention that circles have a center property, rather than
a position.

Also adds a page to the example app to demonstrate its use,
which I tested on iOS and Android.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

File: packages/google_maps_flutter/test/circle_updates_test.dart [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    SystemChannels.platform_views.setMockMethodCallHandler(
+        fakePlatformViewsController.fakePlatformViewsMethodHandler);
+  });
+
+  setUp(() {
+    fakePlatformViewsController.reset();
+  });
+
+  testWidgets('Initializing a circle', (WidgetTester tester) async {
+    final Circle p1 = Circle(circleId: CircleId("circle_1"));
```
Comment (iskakaushik) [inline packages/google_maps_flutter/test/circle_updates_test.dart]: nit: rename p1 to c1 and so on. We've use p1 for polylines, m1 for markers. Let's be consistent here.

File: packages/google_maps_flutter/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## 0.5.10
```
Comment (iskakaushik) [inline packages/google_maps_flutter/CHANGELOG.md]: Please rebase and update the version here and in the `pubspec.yml`.

File: packages/google_maps_flutter/test/circle_updates_test.dart (hunk 1 above)
Comment (matthewlloyd) [reply packages/google_maps_flutter/test/circle_updates_test.dart]: Done.

File: packages/google_maps_flutter/CHANGELOG.md (hunk 2 above)
Comment (matthewlloyd) [reply packages/google_maps_flutter/CHANGELOG.md]: Done.

**Pull request merged**